### PR TITLE
Using JWT auth for local. Generate JWT keys on the fly if not provided.

### PR DESCRIPTION
**Pull request by crufter**

### CODE REVIEW

File: internal/user/user.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+func init() {
+	user, err := user.Current()
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	Dir = filepath.Join(user.HomeDir, path)
+}
+
+func GetKeys() ([]byte, []byte, error) {
```
Comment (crufter) [inline internal/user/user.go:32]: This should not be here as it's server specific. Suggestions welcome

File: internal/user/user.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	"encoding/base64"
+	"encoding/pem"
+	"io/ioutil"
+	"os"
+
+	"github.com/micro/micro/v3/service/logger"
+)
+
+var (
+	Dir  = ""
```
Comment (crufter) [inline internal/user/user.go:20]: Probably should be called microdir

Comment (asim) [reply internal/user/user.go:20]: user.Dir is fine

File: internal/user/user.go (hunk 1 above)
Comment (asim) [reply internal/user/user.go:36]: Don't worry too much for now but can be refactored to internal/server if need be.

**Review by asim — APPROVED** [review]

File: internal/config/config.go [hunk 3]
```diff
@@ -157,7 +152,7 @@ func newConfig() (conf.Config, error) {
 
 	// now write the file if it does not exist
 	if _, err := os.Stat(path); os.IsNotExist(err) {
-		ioutil.WriteFile(path, []byte(`{}`), 0644)
+		ioutil.WriteFile(path, []byte(`{"env":"local"}`), 0644)
```
Comment (crufter) [inline internal/config/config.go:155]: Have noticed `micro user token`returns nothing as env is empty after I nuke `~/.micro`. This fixes that.

File: internal/user/user.go [hunk 4]
```diff
@@ -33,7 +33,10 @@ func init() {
 	}
 }
 
-func GetKeys() ([]byte, []byte, error) {
+// GetJWTCerts returns local keys or generates and returns them for JWT auth.GetJWTCerts
+// This is only here for "0 dep", so people don't have to create and load the certs themselves,
+// not really intended for serious production use.
+func GetJWTCerts() ([]byte, []byte, error) {
```
Comment (asim) [inline internal/user/user.go:39]: The previous name was fine. Its a keypair. So GetKeys made sense. The only thing to do would be to move to an internal/server package.

**Pull request merged**